### ISSUE TITLE
Comment hash populating empty comments

### ISSUE THREAD

**Issue by CMMcEntire** [CMMcEntire/ruby-blog#1]
Deleting them causes the app to complain that there is no route for such an action. Other comments delete fine. 

I tried to add "null: false" to the model fields for commenter and body, but I was unable to test whether this solution worked, because permission to reset the database was denied. 

Permission denied @ apply2files - ./db/development.sqlite3
Couldn't drop database 'db/development.sqlite3'
rake aborted!
Errno::EACCES: Permission denied @ apply2files - ./db/development.sqlite3

**Comment by CMMcEntire** [comment]
Reading the comments helps. 
"Instead of editing this file, please use the migrations feature of Active Record to incrementally modify your database, and then regenerate this schema definition." 

Though, editing the migration file directly does not work either. It gives no errors, but it also makes no changes to the schema. I'll have to figure out how to go through the Action Record to make the changes that I want.

**Comment by CMMcEntire** [comment]
Loading the comments table through the rails console reveals that no nil comments are saved. I created a new comment with nil values and tried to save it, but it was rejected. So this confirms that validation is working correctly! Hooray. 

This also reveals that the likely area where this bug is happening is within the views. That's where I'll go next.

**Comment by CMMcEntire** [comment]
Yep, the issue was in the partial. Before opening this issue I had tried to condition the show article template to display the comments only if there were comments, but it wasn't working. My first condition was always evaluating to true, and always displaying what I didn't want it to display (empty comments). Since this didn't work, I led myself to believe I was in the wrong area. 

To solve the issue, I went into the comments partial and conditioned it to display only if a comment's body wasn't equal to nil. Since validation works, I saw no need to overcomplicate it. Issue closed. :)
